src/platform/datapath_raw_xdp.c

```diff
@@ -1077,88 +1077,93 @@ CxPlatDpRawInitialize(
         Xdp,
         Xdp->WorkerCount);
 
-    PIP_ADAPTER_ADDRESSES Adapters = NULL;
-    ULONG Error;
-    ULONG AdaptersBufferSize = 15000; // 15 KB buffer for GAA to start with.
-    ULONG Iterations = 0;
-    ULONG flags = // skip info that we don't need.
-        GAA_FLAG_INCLUDE_PREFIX |
-        GAA_FLAG_SKIP_UNICAST |
-        GAA_FLAG_SKIP_ANYCAST |
-        GAA_FLAG_SKIP_MULTICAST |
-        GAA_FLAG_SKIP_DNS_SERVER |
-        GAA_FLAG_SKIP_DNS_INFO;
-
-    do {
-        Adapters = (IP_ADAPTER_ADDRESSES*)CxPlatAlloc(AdaptersBufferSize, ADAPTER_TAG);
-        if (Adapters == NULL) {
+    PMIB_IF_TABLE2 pIfTable;
+    if (GetIfTable2(&pIfTable) != NO_ERROR) {
+        Status = QUIC_STATUS_INTERNAL_ERROR;
+        goto Error;
+    }
+
+    struct IfMatching {
+        MIB_IF_ROW2* IfRow;
+        MIB_IF_ROW2* ActualIfRow; // for VF
+    };
+
+    struct IfMatching IfMatching[16] = {0}; // TODO: linked list
+    int IfRowsCount = 0;
+    for (int i = 0; i < (int) pIfTable->NumEntries; i++) {
+        MIB_IF_ROW2* pIfRow = &pIfTable->Table[i];
+        if (pIfRow->InterfaceAndOperStatusFlags.FilterInterface ||
+            !pIfRow->InterfaceAndOperStatusFlags.HardwareInterface ||
```

Comment:
Do we need the `HardwareInterface` or `ConnectorPresent` checks?

Reply:
Empirically these three flags are useful for filter out useless interfaces. Not sure other corner case